test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,290 @@
+    const auto hash = TestTransactionBuilder()
+                          .creatorAccountId("account" + std::to_string(i))
+                          .build()
+                          .hash();
+    tx_request.set_tx_hash(shared_model::crypto::toBinaryString(hash));
+
+    iroha::protocol::ToriiResponse toriiResponse;
+    std::shared_ptr<shared_model::interface::TransactionResponse> response =
+        status_factory->makeEnoughSignaturesCollected(hash, "");
+
+    EXPECT_CALL(*command_service, getStatus(hash)).WillOnce(Return(response));
+
+    transport_grpc->Status(&context, &tx_request, &toriiResponse);
+
+    ASSERT_EQ(toriiResponse.tx_status(),
+              iroha::protocol::TxStatus::ENOUGH_SIGNATURES_COLLECTED);
+  }
+}
+
+/**
+ * @given torii service and number of transactions
+ * @when calling ListTorii
+ * @then ensure that CommandService called handleTransactionBatch as the tx num
+ */
+TEST_F(ToriiTransportCommandTest, ListTorii) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request;
+  std::vector<shared_model::interface::types::HashType> tx_hashes;
+  for (size_t i = 0; i < kTimes; ++i) {
+    auto shm_tx = shared_model::proto::TransactionBuilder()
+                      .creatorAccountId("doge@master" + std::to_string(i))
+                      .createdTime(iroha::time::now())
+                      .setAccountQuorum("doge@master", 2)
+                      .quorum(1)
+                      .build()
+                      .signAndAddSignature(
+                          shared_model::crypto::DefaultCryptoAlgorithmType::
+                              generateKeypair())
+                      .finish();
+    tx_hashes.push_back(shm_tx.hash());
+    new (request.add_transactions())
+        iroha::protocol::Transaction(shm_tx.getTransport());
+  }
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(kTimes);
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service and number of invalid transactions
+ * @when calling ListTorii
+ * @then ensure that CommandService haven't called handleTransactionBatch
+ *       and StatusBus update status tx num times
+ */
+TEST_F(ToriiTransportCommandTest, ListToriiInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request;
+  std::vector<shared_model::interface::types::HashType> tx_hashes;
+  for (size_t i = 0; i < kTimes; ++i) {
+    auto shm_tx = TestTransactionBuilder().build();
+    tx_hashes.push_back(shm_tx.hash());
+    new (request.add_transactions())
+        iroha::protocol::Transaction(shm_tx.getTransport());
+  }
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(0);
+  EXPECT_CALL(*status_bus, publish(_)).Times(kTimes);
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service
+ *        and some number of valid transactions
+ *        and one stateless invalid tx
```

Comment:
There are should be two different test: one for valid transaction and another one for invaild

Reply:
There are already. This test cover different cases, where both types of transaction is present in the list.